git.go

```diff
@@ -251,3 +277,63 @@ func SetupRepoAuth(options *Options) transport.AuthMethod {
 	}
 	return auth
 }
+
+// FetchCoderSSHKeyRetry wraps FetchCoderSSHKey in backoff.Retry.
+// Retries are attempted if Coder responds with a 401 Unauthorized.
+// This indicates that the workspace build has not yet completed.
+// It will retry for up to 1 minute with exponential backoff.
+// Any other error is considered a permanent failure.
+func FetchCoderSSHKeyRetry(ctx context.Context, log LoggerFunc, coderURL, agentToken string) (gossh.Signer, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	signerChan := make(chan gossh.Signer, 1)
+	eb := backoff.NewExponentialBackOff()
+	eb.MaxElapsedTime = 0
+	eb.MaxInterval = time.Minute
+	bkoff := backoff.WithContext(eb, ctx)
+	err := backoff.Retry(func() error {
+		s, err := FetchCoderSSHKey(ctx, coderURL, agentToken)
+		if err != nil {
+			var sdkErr *codersdk.Error
+			if errors.As(err, &sdkErr) && sdkErr.StatusCode() == http.StatusUnauthorized {
+				// Retry, as this may just mean that the workspace build has not yet
+				// completed.
+				log(codersdk.LogLevelInfo, "#1: 🕐 Backing off as the workspace build has not yet completed...")
+				return err
+			}
+			close(signerChan)
```

Comment:
review: I'm open to adding more cases here, but keeping the retry scope small for now.